### PR TITLE
IS-13866 - Merge connector cli with sesam-py

### PR DESCRIPTION
**Pull request by parhambarazesh**
This PR merges the connector cli tool with sesam-py.
when running the "upload" and "download" commands, sesam-py checks if the working directory is a connector and expands/collapses the configs respectively.

The updated code can work in multiple scenarios:
1. Running the commands with the connector (e.g. hubspot connector) set as the working directory.
example params: -vv --system-placeholder testplacholder -e .expanded_directory

2. Running the commands from the root with connector directory name passed as an argument.
example params: upload -vv --system-placeholder testplacholder -d hubspot-connector -e .expanded_directory

4. Running the commands from the root without passing a connector as arg (plane sesam directory)

the connector cli arguments are added to sesampy now which include:
--system-placeholder: placeholder for system/pipe config file name (default value is xxxxxx)
-d: name of the connector directory
-e: name of the expanded directory (default value is .expanded)

The readme is updated correspondingly.

UPDATE:
1. Authentication via oauth and tripletex are integrated
2. flask server used for oauth, stops automatically after login
3. readme update

### CODE REVIEW

File: sesam.py [hunk 1]
```diff
@@ -2212,8 +2257,46 @@ def format(self, record):
     parser.add_argument('-force', dest='force', required=False, action='store_true',
                         help="force the command to run (only for 'upload' and 'download' commands) for non-dev "
                              "subscriptions")
+
+    parser.add_argument("--system-placeholder", metavar="<string>",
```
Comment (branislavjenco) [inline sesam.py:2261]: For every one of these new arguments, I would add a note that they are relevant only when working on connectors to their `help` string.

File: sesam.py [hunk 2]
```diff
@@ -2212,8 +2257,46 @@ def format(self, record):
...
+
+    parser.add_argument("--days", metavar="<string>",
+                        type=int, default=10, help="number of days until the token should expire")
+
+    parser.add_argument("--login_service", metavar="<string>",
+                        type=str, default="oauth", help="login service to use", choices=["oauth", "tripletex"])
+
     try:
         args = parser.parse_args()
+        args.use_connector = os.path.exists("manifest.json") or os.path.exists(os.path.join(args.connector_dir, "manifest.json"))
```
Comment (branislavjenco) [inline sesam.py:2299]: I would change the name to `is_connector` as it's about whether "are we working on a connector repository". :)

File: readme.usage.md [hunk 3]
```diff
@@ -18,7 +18,28 @@ placing it into the topmost parent directory of your repo.
 
 Optionally, you can use another filename and location, and then specify it as a command line argument.
 
-#### 2- sesamconfig
+#### 2- authconfig
+
+In case of using a connector (e.g. Hubspot connector), Sesam client uses `.authconfig` file to authenticate via oauth or
```
Comment (branislavjenco) [inline readme.usage.md:23]: I'd just adjust the text slightly - it's when you are working on a connector repository, not really "using a connector". So I'd rephrase it from `In case of using a connector` to `When working on a connector`, or something like that.

File: readme.usage.md [hunk 4]
```diff
@@ -18,7 +18,28 @@ placing it into the topmost parent directory of your repo.
 
 Optionally, you can use another filename and location, and then specify it as a command line argument.
 
-#### 2- sesamconfig
+#### 2- authconfig
+
+In case of using a connector (e.g. Hubspot connector), Sesam client uses `.authconfig` file to authenticate via oauth or 
+Tripletex and update secrets and environmental variables. To utilize authconfig create `.authconfig` in the working
```
Comment (branislavjenco) [inline readme.usage.md:24]: Absolute super nitpick - environment variables, not environmental :)

File: readme.usage.md [hunk 5]
```diff
@@ -61,6 +82,7 @@ usage: sesam [-h] [-version] [-v] [-vv] [-vvv] [-skip-tls-verification] [-sync-c
              [command]
 
 Commands:
+  connect   Connects to the target node via Oauth or Tripletex authentication and updates secrets and environmental variables.
```
Comment (branislavjenco) [inline readme.usage.md:85]: I suggest we call this command `authenticate` as it's really about authenticating against some external service and getting the necessary secrets that are then uploaded into the node - it's not about connecting or authenticating with the Sesam node. 

Thus I would change the description as well:
"Authenticates against the external service of the connector and updates secrets and environment variables (only when working on a connector).

File: sesam.py (hunk 2 above)
Comment (parhambarazesh) [reply sesam.py:2299]: It makes sense. we can use os.path.exists(os.path.join(args.connector_dir, "manifest.json")) and this will cover both cases. Just in case of wrongly passing connector_dir as arg while the workdir is the connector, this will be false while the combined condition will still pass. Maybe that case is negligible and we can just use the simpler condition since it is incorrect user input.

Btw, it's good that you gave a comment here. I think I should have used os.path.isfile instead. That makes more sense. I will change that too.

Comment (branislavjenco) [reply sesam.py:2299]: I see what you mean with the edge case - up to you, it's not bad with the `or` condition either. It's really a question of should a "wrong" user input still work or not. You could argue that we should let the user know that their input is wrong (i.e. both setting the arg to something and having the manifest in the workdir).

File: sesam.py (hunk 1 above)
Comment (parhambarazesh) [reply sesam.py:2261]: done

File: readme.usage.md (hunk 3 above)
Comment (parhambarazesh) [reply readme.usage.md]: done

File: readme.usage.md (hunk 4 above)
Comment (parhambarazesh) [reply readme.usage.md]: done:)

File: readme.usage.md [hunk 6]
```diff
@@ -18,7 +18,28 @@ placing it into the topmost parent directory of your repo.
...
+CLIENT_ID=<client id for oauth authentication>
+CLIENT_SECRET=<client secret for oauth authentication>
+```
+for Tripletex authentication:
+```bash
+CONSUMER_TOKEN=<client id for Tripletex authentication>
+EMPLOYEE_TOKEN=<client secret for Tripletex authentication>
+```
+
+* The secrets and environmental variables will be updated in the node specified in the syncconfig file.
```
[the remark this answers is not in the document]
Comment (parhambarazesh) [reply readme.usage.md]: done

File: readme.usage.md (hunk 5 above)
Comment (parhambarazesh) [reply readme.usage.md]: done

File: readme.usage.md [hunk 7]
```diff
@@ -133,6 +155,30 @@ optional arguments:
                         use with the init command to add test entities to input pipes
   -force-add
                         use with the '-add-test-entities' option to overwrite test entities that exist locally
+  --system-placeholder <string>
```
[the remark this answers is not in the document]
Comment (parhambarazesh) [reply readme.usage.md:158]: done

File: sesam.py (hunk 2 above)
Comment (parhambarazesh) [reply sesam.py]: For now I have removed the other condition. Then we can discuss that. Maybe it's good to log the is_connector so the user will get informed about the status. We can have it as a small change in another PR.

**Review by branislavjenco — APPROVED** [review]
